lantiq/xrx200: make use of WLAN button on Fritzbox 3370

dzzinstant opened the pull request:
Currently, pressing the WLAN button on the 3370 has no effect.
This submission fixes the GPIO <-> key mapping for this button:
A sequence of pressing and releasing the button will now toggle wireless activity, as would be expected.

Note:
I expect the same problem on Fritzbox 736x devices, but couldn't test this.

dzzinstant committed:
lantiq/xrx200: make WLAN button responsive on Fritzbox 3370

Pressing the 'WLAN' button should enable/disable wireless activity.
However, on the Fritzbox 3370 this doesn't have an effect.
This patch changes the mapping of the physical WLAN button, so a button
press will emit an action for the 'rfkill' key instead of 'wlan'.
Apparently, this is what stock OpenWRT expects, and also what is
implemented for most other devices.

Signed-off-by: Dustin Gathmann <dzsoftware@posteo.org>

dzzinstant committed:
lantiq/xrx200: fix WLAN button actions for Fritzbox 3370

The WLAN button actions are reversed, i.e. pressing the button emits a
'released' action, and vice versa.
This can easily be checked by adding
logger -t button_action "$BUTTON $ACTION"
as the second line of /etc/rc.button/rfkill, and using logread to read
the events (assuming the preceding patch has been applied).
Defining the GPIO as ACTIVE_LOW corrects this behavior.

Signed-off-by: Dustin Gathmann <dzsoftware@posteo.org>

adrianschmutzler commented:
@dzzinstant Can you please update your commits to feature the full first name in the Signed-off-by, I will merge them then.

Despite, since KEY_WLAN vs. KEY_RFKILL is Openwrt behavior, I'd merge this for 736x as well if I receive a patch (you may just push it here as a third commit if you like)

adrianschmutzler commented:
@dzzinstant Ping.

dzzinstant committed:
lantiq/xrx200: make WLAN button responsive on Fritzbox 7360 & 7362

Pressing the 'WLAN' button should enable/disable wireless activity.
Currently, the button is mapped to the KEY_WLAN, which will not
have this effect.
This patch changes the mapping of the WLAN button, so a button
press will emit an action for the 'rfkill' key instead of 'wlan'.
Apparently, this is what stock OpenWRT expects.

This fix is analogous to the preceding patch for Fritzbox 3370.

Signed-off-by: Dustin Gathmann <dzsoftware@posteo.org>

adrianschmutzler commented:
Thanks, merged.

dzzinstant commented:
Thanks! Sorry for my late reaction.
